app/views/questionnaire.py

```diff
@@ -303,20 +353,59 @@ def submit_answers(routing_path, eq_id, form_type):
         if not sent:
             raise SubmissionFailedException()
 
-        get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
+        submitted_time = datetime.utcnow()
 
-        session_store = get_session_store()
-        session_store.session_data.submitted_time = datetime.utcnow().isoformat()
-        session_store.save()
+        _store_session_data(submitted_time)
+        _store_viewable_submission(answer_store.answers, metadata, submitted_time)
 
-        return redirect(url_for(
-            'post_submission.get_thank_you',
-            eq_id=eq_id,
-            form_type=form_type))
+        get_questionnaire_store(current_user.user_id, current_user.user_ik).delete()
+
+        return redirect(url_for('post_submission.get_thank_you', eq_id=eq_id, form_type=form_type))
     else:
         return redirect(invalid_location.url(metadata))
 
 
+def _store_session_data(submitted_time):
+    session_store = get_session_store()
+    session_data = session_store.session_data
+    session_data.submitted_time = submitted_time.isoformat()
+    session_store.save()
+
+
+def _store_viewable_submission(answers, metadata, submitted_time):
+    if is_view_submitted_response_enabled(g.schema.json):
```

Comment:
I think it would be clearer if the if statement was move outside of this method, to make it clear we only store the viewable submission if the schema enables it.